src/components/SigningModals/TransactionStepper/useStepperLayout.js

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useRef, useState } from 'react'
+import useMeasure from 'react-use-measure'
+
+function useStepperLayout() {
+  const [outerBoundsRef, outerBounds] = useMeasure()
+  const innerBoundsRef = useRef()
+  const [innerBounds, setInnerBounds] = useState(null)
+  const [measuring, setMeasuring] = useState(true)
+
+  // First render must always be in "multiple" mode so that our measurement reference point is accurate
+  const [layout, setLayout] = useState('multiple')
+
+  useEffect(() => {
+    const outerMeasured = outerBounds.width && outerBounds.width > 0
+
+    if (!innerBounds) {
+      setInnerBounds(innerBoundsRef.current.offsetWidth)
+      setMeasuring(false)
+    }
+
+    if (outerMeasured && outerBounds.width < innerBounds) {
+      setLayout('single')
+    }
+
+    if (outerMeasured && outerBounds.width >= innerBounds) {
+      setLayout('multiple')
+    }
+  }, [outerBounds, innerBounds])
+
+  return { outerBoundsRef, innerBoundsRef, layout, measuring }
+}
```

Comment:
`use-viewport` doesn't currently provide the ability to contextually measure elements against each other. I know it's on the list but to keep this moving I added `use-measure` for now. Ideally we can come back around and swap this out for an implementation provided by `use-viewport` after it's been added.